libs/common/src/blockchain/blockchain.service.ts

```diff
@@ -403,4 +419,73 @@ export class BlockchainService implements OnApplicationBootstrap, OnApplicationS
     this.logger.debug(`Found tx receipt: ${JSON.stringify(result)}`);
     return result ?? { found: false, success: false };
   }
+
+  /**
+   * Handles the result of a SIWF transaction by extracting relevant values from the transaction events.
+   * @param txResultEvents - The transaction result events to process.
+   * @returns An object containing the extracted SIWF transaction values.
+   */
+  public handleSIWFTxnResult = (txResultEvents: Vec<EventRecord>): SIWFTxnValues => {
+    const siwfTxnValues: Partial<SIWFTxnValues> = {};
+
+    txResultEvents.forEach((record) => {
+      if (record.event && this.api.events.msa.MsaCreated.is(record.event)) {
+        siwfTxnValues.msaId = record.event.data.msaId.toString();
+        siwfTxnValues.address = record.event.data.key.toString();
+      }
+      if (record.event && this.api.events.handles.HandleClaimed.is(record.event)) {
+        const handleHex = record.event.data.handle.toString();
+        // Remove the 0x prefix from the handle and convert the hex handle to a utf-8 string
+        const handleData = handleHex.slice(2);
+        siwfTxnValues.handle = Buffer.from(handleData.toString(), 'hex').toString('utf-8');
+      }
+
+      if (record.event && this.api.events.msa.DelegationGranted.is(record.event)) {
+        siwfTxnValues.newProvider = record.event.data.providerId.toString();
+        const owner = record.event.data.delegatorId.toString();
+        if (owner !== siwfTxnValues.msaId) {
+          throw new Error(`DelegationGranted event owner ${owner} does not match msaId ${siwfTxnValues.msaId}`);
+        }
+      }
```

Comment:
nit: non-blocking, but this would be more effiicient using `else if ...` instead of serial `if ...` blocks

also, I guess this is OK: as long as the we're processing the events in the order they occur in the block, there should be no risk of processing the `DelegationGranted` before the `MsaCreated`--relying on the caller not to have messed with the order of the event records.